aci/resource_aci_cloudtemplateextnetwork.go

```diff
@@ -0,0 +1,449 @@
+	log.Printf("[DEBUG] %s: Beginning Import", d.Id())
+	aciClient := m.(*client.Client)
+	dn := d.Id()
+	cloudtemplateExtNetwork, err := getRemoteCloudTemplateforExternalNetwork(aciClient, dn)
+	if err != nil {
+		return nil, err
+	}
+	schemaFilled, err := setCloudTemplateforExternalNetworkAttributes(cloudtemplateExtNetwork, d)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("[DEBUG] Begining Import of cloud Regions attributes.")
+	regionsData, err := aciClient.ListCloudProviderandRegionNames(cloudtemplateExtNetwork.DistinguishedName)
+	if err != nil {
+		log.Printf("[DEBUG] Error while importing cloud Regions attributes %v", err)
+	}
+
+	RegionsList := make([]string, 0, 1)
+	for _, regionValue := range regionsData {
+		regionsMap, err := setCloudProviderandRegionNamesAttributes(regionValue, make(map[string]string))
+		if err != nil {
+			d.SetId("")
+			return nil, err
+		}
+		RegionsList = append(RegionsList, regionsMap["region"])
+		d.Set("cloud_vendor", regionsMap["cloud_vendor"])
+		if regionsMap["cloud_vendor"] != "aws" {
+			d.Set("router_type", "")
+		}
+	}
+	log.Printf("[DEBUG] : Import cloud regions finished successfully")
+	d.Set("regions", RegionsList)
+
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+	return []*schema.ResourceData{schemaFilled}, nil
+}
+
+func resourceAciCloudTemplateforExternalNetworkCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] CloudTemplateforExternalNetwork: Beginning Creation")
+	aciClient := m.(*client.Client)
+	name := d.Get("name").(string)
+	cloudVendor := d.Get("cloud_vendor").(string)
+	CloudInfraNetworkTemplateDn := "uni/tn-infra/infranetwork-default"
+
+	cloudtemplateExtNetworkAttr := models.CloudTemplateforExternalNetworkAttributes{}
+
+	nameAlias := ""
+	if NameAlias, ok := d.GetOk("name_alias"); ok {
+		nameAlias = NameAlias.(string)
+	}
+
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		cloudtemplateExtNetworkAttr.Annotation = Annotation.(string)
+	} else {
+		cloudtemplateExtNetworkAttr.Annotation = "{}"
+	}
+
+	if Name, ok := d.GetOk("name"); ok {
+		cloudtemplateExtNetworkAttr.Name = Name.(string)
+	}
+
+	if AllRegions, ok := d.GetOk("all_regions"); ok {
+		cloudtemplateExtNetworkAttr.AllRegion = AllRegions.(string)
+		if AllRegions.(string) == "yes" {
+			if cloudVendor == "aws" {
+				if RouterType, ok := d.GetOk("router_type"); ok {
+					if RouterType == "c8kv" {
+						cloudtemplateExtNetworkAttr.HostRouterName = "default"
+					} else {
+						// object class cloudGatewayRouterP
+						if HubNetworkName, ok := d.GetOk("hub_network_name"); ok {
+							cloudtemplateExtNetworkAttr.HubNetworkName = HubNetworkName.(string)
+						}
+					}
+				}
+			} else {
+				// Always true for Azure cloud
+				cloudtemplateExtNetworkAttr.HostRouterName = "default"
+			}
+		} else {
+			// following 2 attributes are used only in GCP
```

Comment:
why we don't need cloudVendor == 'gcp' as condition

Reply:
AWS uses both parameters that are Azure and GCP specific, adding a condition will make it difficult for AWS cloud usage.